go-auth.go

```diff
@@ -697,10 +781,12 @@ func CheckBackendsAcl(username, topic, clientid string, acc int) bool {
 			}
 			var backend = authPlugin.backends[bename]
 			log.Debugf("Superuser check with backend %s", backend.GetName())
-			if backend.GetSuperuser(username) {
+			if ok, err := backend.GetSuperuser(username); ok && err == nil {
 				log.Debugf("superuser %s acl authenticated with backend %s", username, backend.GetName())
 				aclCheck = true
 				break
+			} else if err != nil && firstError == nil {
```

Comment:
No need for `else if` if you're breaking in the previous clause.

Reply:
I need an `else if` because the `err` variable (now renamed to `getSuperuserErr`) is only defined in the if/else scope.